internal/engine/image_build_and_deployer.go

```diff
@@ -211,6 +211,7 @@ func (ibd *ImageBuildAndDeployer) deploy(ctx context.Context, st store.RStore, p
 	deployLabel := k8s.TiltDeployLabel(deployID)
 
 	var targetIDs []model.TargetID
+	injectedSynclet := false
```

Comment:
i don't think this will work correctly in the case where you have multiple pods? is that a case we care about?

Reply:
it might be good to map this by entity name/kind or smth

Reply:
@nicks what do you think of this approach where we only inject a synclet at most once for each Kubernetes entity?

Also can you say more about multiple pods? You mean replica sets? Or multiple deployments?